addons/point_of_sale/wizard/pos_payment.py

```diff
@@ -53,10 +54,11 @@ def check(self):
         currency = order.currency_id
 
         init_data = self.read()[0]
+        payment_method = self.env['pos.payment.method'].browse(init_data['payment_method_id'][0])
         if not float_is_zero(init_data['amount'], precision_rounding=currency.rounding):
             order.add_payment({
                 'pos_order_id': order.id,
-                'amount': order._get_rounded_amount(init_data['amount']),
+                'amount': order._get_rounded_amount(init_data['amount'], payment_method.is_cash_count),
```

Comment:
It's not only for cash, right? We also allow payment rounding on other payment methods. So I guess the boolean to pass here should be based on the cash_rounding config and the type of payment method.

Reply:
`_get_rounded_amount` already check if the config use rounding, so I just made sure we round all methods if the `only_round_cash_method` is not activated.